org-code-javabuilder/theater/src/main/java/org/code/theater/Prompter.java

```diff
@@ -1,11 +1,22 @@
 package org.code.theater;
 
+import java.util.HashMap;
 import org.code.media.Image;
+import org.code.protocol.GlobalProtocol;
+import org.code.protocol.OutputAdapter;
 
 public class Prompter {
-  protected Prompter() {}
+  private final OutputAdapter outputAdapter;
+
+  protected Prompter() {
+    this.outputAdapter = GlobalProtocol.getInstance().getOutputAdapter();
+  }
```

Comment:
nit: might want to have a second constructor that takes in `outputAdapter` as a parameter and call that here, for testing purposes (similar to `Board` and `Stage`) so you don't have to mock the global protocol instance

Reply:
That does make things a lot cleaner, thanks